kusto/ingest/status.go

```diff
@@ -0,0 +1,291 @@
+		OperationID:                uuid.Nil,
+		ActivityID:                 uuid.Nil,
+		ErrorCode:                  "Unknown",
+		FailureStatus:              Unknown,
+		Details:                    "",
+		OriginatesFromUpdatePolicy: false,
+	}
+
+	return rec
+}
+
+// FromProps takes in data from ingestion options.
+func (r *StatusRecord) FromProps(props properties.All) {
+	r.IngestionSourceID = props.Source.ID
+	r.Database = props.Ingestion.DatabaseName
+	r.Table = props.Ingestion.TableName
+	r.UpdatedOn = time.Now()
+
+	if props.Ingestion.BlobPath != "" && r.IngestionSourcePath == "Undefined" {
+		r.IngestionSourcePath = props.Ingestion.BlobPath
+	}
+}
+
+// FromMap converts an ingestion status record to a key value map.
+func (r *StatusRecord) FromMap(data map[string]interface{}) {
+	if data["Status"] != nil {
+		r.Status = StatusCode(data["Status"].(string))
+	}
+
+	if data["IngestionSourceId"] != nil {
+		uid, err := getGoogleUUIDFromInterface(data["IngestionSourceId"])
+		if err == nil {
+			r.IngestionSourceID = uid
+		}
+	}
+
+	if data["IngestionSourcePath"] != nil {
+		r.IngestionSourcePath = data["IngestionSourcePath"].(string)
+	}
+
+	if data["Database"] != nil {
+		r.Database = data["Database"].(string)
+	}
+
+	if data["Table"] != nil {
+		r.Table = data["Table"].(string)
+	}
+
+	if data["UpdatedOn"] != nil {
+		t, err := getTimeFromInterface(data["UpdatedOn"])
+		if err == nil {
+			r.UpdatedOn = t
+		}
+	}
+
+	if data["OperationId"] != nil {
+		uid, err := getGoogleUUIDFromInterface(data["OperationId"])
+		if err == nil {
+			r.OperationID = uid
+		}
+	}
+
+	if data["ActivityId"] != nil {
+		uid, err := getGoogleUUIDFromInterface(data["ActivityId"])
+		if err == nil {
+			r.ActivityID = uid
+		}
+	}
+
+	if data["ErrorCode"] != nil {
+		r.ErrorCode = data["ErrorCode"].(string)
+	}
+
+	if data["FailureStatus"] != nil {
+		r.FailureStatus = FailureStatusCode(data["FailureStatus"].(string))
+	}
+
+	if data["Details"] != nil {
+		r.Details = data["Details"].(string)
+	}
+
+	if data["OriginatesFromUpdatePolicy"] != nil {
+		r.OriginatesFromUpdatePolicy = data["OriginatesFromUpdatePolicy"].(bool)
+	}
+
+}
+
+// ToMap converts an ingestion status record to a key value map.
+func (r *StatusRecord) ToMap() map[string]interface{} {
+	data := make(map[string]interface{})
+
+	// Since we only create the initial record, It's not our responsibility to write the following fields:
+	//   OperationID, AcitivityID, ErrorCode, FailureStatus, Details, OriginatesFromUpdatePolicy
+	// Those will be read from the server if they have data in them
+	data["Status"] = r.Status
+	data["IngestionSourceId"] = r.IngestionSourceID
+	data["IngestionSourcePath"] = r.IngestionSourcePath
+	data["Database"] = r.Database
+	data["Table"] = r.Table
+	data["UpdatedOn"] = r.UpdatedOn.Format(time.RFC3339Nano)
+
+	return data
+}
+
+// String converts an ingestion status record a printable  string.
+func (r *StatusRecord) String() string {

```

Comment:
I'd probably just do:

return pretty.Sprint(r)

Think that will not only get you a better result, but if we add fields we don't have to go back and update this.